### DISCUSSION TITLE
remove certain fields from whatever decides `form.valid`

### DISCUSSION THREAD

**Discussion by lifeiscontent** [edmundhung/conform#630] [Ideas]
in the scenario that I want to disable submit buttons if the form is not valid, I can't currently do that with a schema that uses `intent` to discriminate between the actions

e.g.

```tsx

const todoSchema = z.object({
   id: z.string().min(1),
   title: z.string().min(1),
  completed: z.boolean(),
})

const schema = z.discriminatedUnion("intent", [
  z.object({
    intent: z.literal("createTodo"),
    title: z.string().min(1),
    completed: z.boolean(),
  }),
  z.object({
    intent: z.literal("updateTodo"),
    id: z.string().min(1),
    title: z.string().min(1),
    completed: z.boolean(),
  }),
  z.object({
    intent: z.literal("deleteTodo"),
    id: z.string().min(1),
  }),
]);
```

because `intent` is going to be a value on `<button />` `intent` would never validate unless it came from the button itself directly

it would be nice to have something like shouldDirtyConsider, except for skipping validation in conform entirely

**Comment by edmundhung** [comment]
Value only exists on the button is tricky 🤔 But can you put the intent in a hidden input instead of the button?

**Reply by lifeiscontent** [reply]
@edmundhung that's not a bad idea, that might solve a number of use cases, e.g. having to maintain a "currentIntent" to render inside of a hidden button for submissions that are coming from input, I'll give it a try today and report back if I run into any issues

**Reply by lifeiscontent** [reply]
@edmundhung seems to work! Thank you 🙏🏻
